MapVars

Hydraxous opened the pull request:
Makes MapVars fully functional. (Presumably how PITR intended them to work)

Note: I've set up the save location to be similar to how the vanilla MapVar system would do it so it will work with save slots, but with some distinction specifically for Angry. I figured you may want to change this, so [here](https://github.com/Hydraxous/AngryLevelLoader/blob/3e3a91df3c5975d10ce62bdacc3bd8ca090066e9/AngryLevelLoader/Managers/AngryMapVarManager.cs#L23).

I know you mentioned adding some custom stuff on top of this so if you want me to integrate that too I can do that.

Hydraxous committed:
patch mapvarmanager

Hydraxous committed:
implement mapvars

Hydraxous committed:
Merge remote-tracking branch 'upstream/release' into persistent-mapvar-fix

Hydraxous committed:
add call for debugging, small optimization

Hydraxous committed:
fix vars persisting, fix vars not loading after restart

Hydraxous committed:
Update file structure

Hydraxous committed:
fix path error

eternalUnion commented:
There are some logical problems with it
- If user changes preset (plugin configurator), persistent vars should be changed as well. This means that there should somehow exist multiple copies of the persistent file for each preset (I will try to implement this myself)
- On preset reset, level ranks and secrets will be removed but persistent vars will live on, which could cause undefined behavior

Hydraxous committed:
update paths to use Config folder

Gonna rewrite a lot in a non singleton class to allow for custom stores and stuff

Hydraxous committed:
fix nfe from switching preset in main menu

Hydraxous committed:
change code to instancing use of MapVarHandler

Hydraxous committed:
fix file name discrepancy

Hydraxous committed:
Update AngryMapVarManager.cs

Hydraxous committed:
add note

Hydraxous committed:
remove testing block

Hydraxous committed:
Merge remote-tracking branch 'upstream/release' into persistent-mapvar-fix

Hydraxous committed:
Update AngrySceneManager.cs
